core/src/test/java/com/linecorp/armeria/internal/common/SchemeAndAuthorityTest.java

```diff
@@ -45,8 +47,15 @@ void fromAuthority(String authority, String expectedAuthority, String expectedHo
 
     @ParameterizedTest
     @CsvSource({
-            "foo:bar", "http://foo:80", "foo/bar", "foo?bar=1", "foo#bar",
-            "[192.168.0.1]", "[::1", "::1]", "[::1]%eth0", "unix:foo.sock"
+            "foo:bar",        // Invalid port
+            "http://foo:80",  // Scheme included
+            "foo/bar",        // Authority with path
+            "foo?bar=1",      // Authority with query
+            "foo#bar",        // Authority with fragment
+            "[192.168.0.1]",  // Bracketed IPv4
+            "[::1", "::1]",   // Incomplete IPv6
+            "[::1]%eth0",     // IPv6 with scope
```

Comment:
I think an IPv6 address with scope is represented like this:

- `[::1%eth0]:8080`

In this case, we should simply strip the scope and normalize into:

- `[::1]:8080`

However, `[::1]%eth0` should remain failing.

Reply:
TIL 📝